src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs

```diff
@@ -98,14 +103,23 @@ internal static SyntaxNode TryGetCorrespondingLambdaBody(SyntaxNode oldBody, Syn
                         (IsReducedSelectOrGroupByClause(newGroup, newGroup.GroupExpression) ? null : newGroup.GroupExpression) : newGroup.ByExpression;
 
                 case SyntaxKind.LocalFunctionStatement:
-                    var newLocalFunction = (LocalFunctionStatementSyntax)newLambda;
-                    return (SyntaxNode)newLocalFunction.Body ?? newLocalFunction.ExpressionBody;
+                    return GetLocalFunctionBody((LocalFunctionStatementSyntax)newLambda);
 
                 default:
                     throw ExceptionUtilities.UnexpectedValue(oldLambda.Kind());
             }
         }
 
+        public static SyntaxNode TryGetCorrespondingLambdaBody(SyntaxNode lambdaExpression)
```

Comment:
OK

---
In reply to: [134012659](https://github.com/dotnet/roslyn/pull/21599#discussion_r134012659) [](ancestors = 134012659)